docs(): interop docs

kelemeno opened the pull request:
## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

kelemeno committed:
interop docs

kelemeno committed:
duplicate code

kelemeno committed:
trailing white space

kelemeno committed:
readd some deleted files

0xValera committed:
added full contracts tree into root SUMMARY.md

0xValera committed:
fixed contracts tree, fixed links leading outside of docs directory, lint

1. Changed links leading outside the docs directory to fixed commit links. Otherwise mdbook doesn't compile.
For genesis generation section, main branch of zksync-era is used.
For contracts commit b43cf6b of era-contracts repo is used. Links that lead to other github commits left unchanged.
In cross_chain_paymaster.md commit f4ebeec of zksync-era repo is used.
2. Created a file for each of the section points that did not have it already.
3. Deleted redundant [back to readme], since now docs will be in website format.

0xValera committed:
linter

0xValera committed:
merge processing_of_l1-l2_txs and l1_to_l2. Lint

0xValera committed:
Merge branch 'main' into kl/interop-docs

zkzoomer committed:
flip ChainBatchRoot children

zkzoomer committed:
flip ChainBatchRoot children

zkzoomer committed:
flip GW ChainBatchRoot children

zkzoomer committed:
typos

zkzoomer committed:
reflect sdk target labels

zkzoomer committed:
fixes

kelemeno committed:
more explanation of standard trigger account and interop handler

0xValera committed:
synchronize Stas's renamings related to L1<>L2 messaging via gateway

0xValera committed:
L3 -> L2 in AssetTracker

0xValera committed:
synchronize Vlad's changes related to L2 DA

kelemeno committed:
shadow account name add

0xValera committed:
docs: Move interop docs + set create-missing to be false in config (#4020)

## What ❔

<!-- What are the changes this PR brings about? -->
Move interop-related docs to be an independent from bridging section in
docs.
Make `create-missing` be equal to false in mdbook config by default.
This way when the link in any of files is leading to an non-existent
file, mdbook will throw an error instead of creating the file. Earlier
it was turned on by default (at least on my side), but now it's required
to pass it in config. We have to have it to ensure that all links lead
to a valid destinations.

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

kelemeno committed:
add create upgrade doc

zkzoomer committed:
interop mode fixes

kelemeno committed:
asset tracker

kelemeno committed:
add interop roots to bootloader doc

kelemeno committed:
Merge branch 'main' of github.com:matter-labs/zksync-era into kl/interop-docs

kelemeno committed:
asset tracker

kelemeno committed:
comment out future version

0xValera committed:
docs: clean the AssetTracker docs (#4335)

kelemeno commented on docs/src/specs/contracts/bridging/asset_tracker/asset_tracker.md:
```diff
@@ -0,0 +1,169 @@
+
+In order for information to flow smoothly, migrating to the Gateway has the following steps:
+
+- Chain's `AssetTracker` gets the `totalSupply` of the token (from the token or from storage) and sends a message to L1.
```
The only attack I see at the moment is iterated L1-> GW migration, without migrating back to L1. This might be a problem. 

We need a formal method of looking at it.

maybe: the chainBalance on GW and the chainBalance on L1 always stores the latest totalSupply in the chain + the outstanding L1 withdrawals.

kelemeno committed:
asset tracker update

kelemeno commented on docs/src/specs/contracts/bridging/asset_tracker/asset_tracker.md:
```diff
@@ -0,0 +1,173 @@
+
+The Gateway component as an intermediate settlement layer complicates the setup. L1->GW->L2 deposits, failed deposits, and L2->GW->L1 withdrawals have to be handled.
+
+First of all for chains settling on Gateway, the balance of the chain in the L1 `AssetTracker` will be moved to the Gateway's `chainBalance`. This is a complicated migration process, see [below](#migrating-and-settling-on-gateway). Similarly, when a chain migrates back to L1, its balance will be separate from the Gateway's `chainBalance`. On Gateway the `chainBalance`s are tracked for each chain  ...
```
yes

nikitastupin-matterlabs commented on docs/src/specs/contracts/bridging/asset_tracker/asset_tracker.md:
```diff
@@ -0,0 +1,173 @@
+
+When a chain migrates from L1 to GW or from GW to L1, the `chainBalance` mapping for each token does not get transferred automatically. They can be trustlessly moved by anyone from GW to L1 and vice versa. But our standard tooling will migrate all token balances for best UX.
+
+When a chain is settling on Gateway it calls the `processLogsAndMessages` on settlement. This means that if the token is not yet migrated, and a withdrawal could be processed, the settlement would fail, since the chain's balance would be zero. In order to prevent this, withdrawals and interop will be paused on the chain until the token's balance is migrated.
```
> withdrawals and interop will be paused on the chain until the token's balance is migrated.

This seems like a blocker for becoming a Stage 1 rollup. I do support pausing for migration but we have to keep this in mind.

0xValera committed:
docs: sb/extend auditor docs on AssetTracker component (#4522)

## What ❔

<!-- What are the changes this PR brings about? -->
<!-- Example: This PR adds a PR template to the repo. -->
<!-- (For bigger PRs adding more context is appreciated) -->

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

---------

Co-authored-by: Stanislav Breadless <stanislavbezkor@gmail.com>

StanislavBreadless committed:
remove todo

StanislavBreadless committed:
upd docs

kelemeno committed:
Merge pull request #4529 from matter-labs/sb-upd-docs

Update docs

kelemeno committed:
update upgrade process

0xValera committed:
lint

StanislavBreadless committed:
Updating interop docs to match the newest state (#4682)

## What ❔

This doc does not yet include base token holder. Will be added a bit
later

## Why ❔

<!-- Why are these changes done? What goal do they contribute to? What
are the principles behind them? -->
<!-- The `Why` has to be clear to non-Matter Labs entities running their
own ZK Chain -->
<!-- Example: PR templates ensure PR reviewers, observers, and future
iterators are in context about the evolution of repos. -->

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Operational changes
<!-- Any config changes? Any new flags? Any changes to any scripts? -->
<!-- Please add anything that non-Matter Labs entities running their own
ZK Chain may need to know -->

## Checklist

<!-- Check your PR fulfills the following items. -->
<!-- For draft PRs check the boxes as you complete them. -->

- [ ] PR title corresponds to the body of PR (we generate changelog
entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev
lint`.

---------

Co-authored-by: 0xValera <55665573+0xValera@users.noreply.github.com>

kelemeno committed:
more docs

kelemeno committed:
Merge branch 'kl/interop-docs' of github.com:matter-labs/zksync-era into kl/interop-docs

kelemeno committed:
linting

kelemeno committed:
linting

kelemeno committed:
small fixes

kelemeno committed:
linting

kelemeno committed:
add wip

kelemeno committed:
atomicity doc

kelemeno committed:
images

kelemeno committed:
some more upgrade

kelemeno committed:
atomic interop using l1

kelemeno committed:
linting

kelemeno committed:
more changes

kelemeno committed:
linting

deniallugo-ml commented:
This PR has no movement for a few years. Closing it as  stale
